src/main/java/randoop/main/GenTests.java

```diff
@@ -1335,4 +1359,93 @@ private Path getResourceDirectoryPath(String resourceDirectory) {
+   * @return classnames of dependencies
+   */
+  public static Set<@ClassGetName String> getDependentClassnamesFromClassnames(
+      Set<@ClassGetName String> classnames, AccessibilityPredicate accessibility) {
+    Set<@ClassGetName String> dependenciesClassnames = new TreeSet<>();
+
+    for (String classname : classnames) {
+      try {
+        Class<?> getDependenciesFrom = Class.forName(classname);
+        for (Method method : getDependenciesFrom.getDeclaredMethods()) {
+          addParameterTypesIfShould(method, dependenciesClassnames, accessibility);
+        }
+        for (Constructor<?> constructor : getDependenciesFrom.getConstructors()) {
+          addParameterTypesIfShould(constructor, dependenciesClassnames, accessibility);
+        }
+      } catch (ClassNotFoundException e) {
+        throw new RandoopUsageError(String.format("Cannot load class %s", classname));
+      }
+    }
+    return dependenciesClassnames;
+  }
+
+  /**
+   * Returns names of classes that methods in {@code methodlist} depend on. Does not add
+   * dependencies of methods that should be omitted. Does not add classes that are not accessible or
+   * should be omitted.
+   *
+   * @param accessibilityPredicate an accessibility predicate
+   * @param omitMethods each regex indicates methods that should not be called directly in generated
+   *     tests
+   * @return classnames of dependencies
+   */
+  public static Set<@ClassGetName String> getDependentClassnamesFromMethodList(
+      AccessibilityPredicate accessibilityPredicate, List<Pattern> omitMethods) {
+    if (methodlist == null) {
+      return Collections.emptySet();
+    }
+    Set<@ClassGetName String> classnames = new TreeSet<>();
+    OmitMethodsPredicate omitMethodsPredicate = new OmitMethodsPredicate(omitMethods);
+    ReflectionPredicate reflectionPredicate = new DefaultReflectionPredicate();
+    TypedClassOperationProvider provider = new TypedClassOperationProvider(omitMethodsPredicate);
+    List<TypedClassOperation> operations = null;
+    try {
+      operations =
+          provider.getOperationsFromFile(methodlist, accessibilityPredicate, reflectionPredicate);
+    } catch (SignatureParseException e) {
+      throw new RandoopUsageError(String.format("%nError: parse exception thrown %s%n", e));
+    }
+    for (TypedClassOperation operation : operations) {
+      AccessibleObject accessibleObject = operation.getOperation().getReflectionObject();
+      if (accessibleObject instanceof Executable) {
+        addParameterTypesIfShould(
+            (Executable) accessibleObject, classnames, accessibilityPredicate);
+      }
+    }
+    return classnames;
+  }
+
+  /**
+   * Adds the parameter types of {@code executable} to {@code classnames} if the parameter type is
+   * not String or primitive, should not be omitted, and is accessible by accessibility predicate.
+   *
+   * @param executable a method or constructor
+   * @param classnames collection to add parameter types to
```

Comment:
A method without side effects is easier to understand than one with side effects.  Why not have this method return a list rather than side-effecting an argument?